server/consumer.go

```diff
@@ -2085,6 +2085,16 @@ func (o *consumer) needAck(sseq uint64) bool {
 	var asflr, osseq uint64
 	var pending map[uint64]*Pending
 	o.mu.RLock()
+
+	// Check first if we are filtered, and if so check if this is even applicable to us.
+	if o.isFiltered() && o.mset != nil {
+		subj, _, _, _, err := o.mset.store.LoadMsg(sseq)
+		if err != nil || !o.isFilteredMatch(subj) {
+			o.mu.RUnlock()
```

Comment:
Oops, I see that you already have unlock the consumer line 2087. So move o.mu.RLock() from line 2087 down after the outer if, assuming that o.isFiltered() or o.isFilteredMatch() are not grabbing the lock.

Reply:
I think its ok.

Reply:
Sorry, I read as an RUnlock() above...